### ISSUE TITLE
Argo CD App Updates should re-attempt conflict errors 

### ISSUE THREAD

**Issue by jessesuen** [akuity/kargo#2489]
# Description

A user hit the following error during a promotion:

```
    phase: Errored
      message: 'error executing Argo CD promotion mechanism: error patching Argo CD
        Application "guestbook": failed to apply
        modifications to unstructured object: unable to update sources to
        desired revisions: resource has been modified'
```

`resource has been modified` is a common scenario that Kargo should deal with transparently: by refetching the latest application object, reapplying the modifications, and retrying the update.

NOTE: if the conflict is with the `.operation` field, then we should not blow away what was set. 

# Version

```
v0.8.7
```

# Logs

```
Paste any relevant application logs here.
```

**Comment by jsirianni** [comment]
I am running into this as well, v0.9.1. You can get around it by triggering another deployment with a nw commit / image tag. Would be nice if it could resolve itself.

**Comment by hiddeco** [comment]
This should be solved via #2766 (which is unreleased at the moment).
